### PR TITLE
Abstract grouping heuristic

### PR DESCRIPTION
**Pull request by JamesSmartCell**
Abstract the grouping heuristic to the respective token class. This gives fine control over how tokens are displayed in the full asset card view (```view``` in TokenScript).

### CODE REVIEW

File: app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java [hunk 1]
```diff
@@ -411,7 +412,7 @@ public boolean hasArrayBalance()
     }
 
     @Override
-    public List<BigInteger> getArrayBalance() { return balanceArray; }
```
Comment (James-Sangalli) [inline app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java]: @JamesSmartCell under what circumstance would you want the balance with zeros in the array for 721 tickets?

**Comment by James-Sangalli** [comment]
@JamesSmartCell Added some fixes to this PR because it crashed on transfer and I edited it so that a non zero balances in the array are returned to prevent incorrect balance count on token card

File: app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java (hunk 1 above)
Comment (JamesSmartCell) [reply app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java]: This is for erc875. Most likely it's an artifact from the ticket class.

Comment (JamesSmartCell) [reply app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java]: But it should be in a different PR as it's unrelated to this one

**Comment by James-Sangalli** [comment]
@JamesSmartCell remember to reference the issue: https://github.com/AlphaWallet/alpha-wallet-android/issues/987

**Pull request merged**